src/on.ts

```diff
@@ -104,3 +104,58 @@ export default function on(target: any, type: any, listener: any, capture?: bool
 
 	throw new TypeError('Unknown event emitter object');
 }
+
+/**
+ * Provides a mechanism for listening to the next occurence of events from event
+ * emitters, Evented objects, or DOM nodes.
+ * @param target Target to listen for event on
+ * @param type Event type(s) to listen for; may be strings or extension events
+ * @param listener Callback to handle the event when it fires
+ * @param capture Whether the listener should be registered in the capture phase (DOM events only)
+ * @return A handle which will remove the listener when destroy is called
+ */
+export function once(target: EventTarget, type: string | string[], listener: EventCallback, capture?: boolean): Handle;
+export function once(target: EventEmitter | Evented, type: string | string[], listener: EventCallback): Handle;
+export function once(target: any, type: any, listener: any, capture?: boolean): Handle {
+	let handle = on(target, type, function () {
+		handle.destroy();
+		return listener.apply(this, arguments);
+	}, capture);
+
+	return handle;
+}
+
+/**
+ * Provides a mechanism for creating pausable listeners for events from event emitters, Evented objects, or DOM nodes.
+ * @param target Target to listen for event on
+ * @param type Event type(s) to listen for; may be strings or extension events
+ * @param listener Callback to handle the event when it fires
+ * @param capture Whether the listener should be registered in the capture phase (DOM events only)
+ * @return A handle which will remove the listener when destroy is called
+ */
+export function pausable(target: EventTarget, type: string | string[], listener: EventCallback, capture?: boolean): PausableHandle;
+export function pausable(target: EventEmitter | Evented, type: string | string[], listener: EventCallback): PausableHandle;
+export function pausable(target: any, type: any, listener: any, capture?: boolean): PausableHandle {
+	let paused: boolean;
+
+	let handle = <PausableHandle> on(target, type, function () {
+		if (!paused) {
+			return listener.apply(this, arguments);
+		}
+	}, capture);
+
+	handle.pause = function () {
+		paused = true;
+		return this;
```

Comment:
Good question... of course the interface specifies `PausableHandle`, but should it?  There is no functional value in handle chaining.  Should maybe just be `void`?